Client/IRequest.cs

```diff
@@ -92,6 +92,16 @@ public interface IRequest
         /// <param name="requestCoordinator">The request coordinator (or null to use the default behaviour).</param>
         IRequest WithRequestCoordinator(IRequestCoordinator? requestCoordinator);
 
+        /// <summary>Add a filter for this request.</summary>
+        /// <typeparam name="TFilter">The type of the filter.</typeparam>
+        /// <param name="filter">The filter.</param>
+        /// <param name="removeExisting">Indicates whether existing filters of type TFilter should be removed before adding the filter.</param>
+        IRequest WithFilter<TFilter>(TFilter filter, bool removeExisting = true) where TFilter : IHttpFilter;
```

Comment:
I don't think `removeExisting` should default to true; `WithFilter` removing filters is unintuitive, and there's a lot of cases where we may want multiple filters of the same type.

I'm not necessarily against having the parameter (if it defaults to false), but wouldn't it be cleaner to keep removing & adding separate? So if you wanted to replace filters, you'd just call both so it's explicit and clear:
```
await request
    .WithoutFilter<ErrorFilter>()
    .WithFilter<SomeCustomErrorFilter>();

Reply:
I'll remove the parameter.